base/threadingconstructs.jl

```diff
@@ -85,6 +86,15 @@ function _threadsfor(iter, lbody, schedule)
                 $(esc(lbody))
             end
         end
+        idx = Atomic{UInt}(1)
+        function threadsfor_fun_dynamic()
+            r = range # Load into local variable
+            lenr = length(r)
+            while (i = atomic_add!(idx, UInt(1))) <= lenr
+                local $(esc(lidx)) = @inbounds r[i]
+                $(esc(lbody))
```

Comment:
We will need to come up with a different approach here. We need to avoid generating two copies of the code: the code size is 2^n for nesting depth n. Even though nesting `@threads` loops is not common, avoiding this is the best practice.

Reply:
Shouldn't the compiler be able to constprop the schedule to delete the branch not taken?

Reply:
That said, if `schedule` is a `QuoteNode`, you could check if `schedule.value === :static` or `schedule.value === :dynamic`, and generate code accordingly.

Reply:
Sorry for late reply.
I think the `:dynamic` version is just like the default version which only use multi-threads in thread 1. So I don't think the 2^n thing will happen.
This is why I said "`:dynamic` creates one task per thread **if possible**" in the doc string.